src/runtime/contrib/clml/clml_runtime.cc

```diff
@@ -528,15 +529,22 @@ class CLMLRuntime : public JSONRuntimeBase {
         uint32_t eid = EntryID(nid, 0);
         node_data = data_entry_[eid]->data;
       }
+
       auto clml_tensor = MakeCLMLTensorFromJSONNode(node, layout, dtype, node_data, shape);
+
       this->layer_.storage_map.insert({nid, std::make_pair(clml_tensor, node)});
 
       if ("input" == node.GetOpType()) {
         this->layer_.inputs.insert({nid, this->layer_.storage_map[nid].first});
         // Input copy placeholder Tensor
-        this->layer_.in_placeholder.insert(
-            {nid, MakeCLMLTensorFromJSONNode(node, CL_TENSOR_LAYOUT_NCHW_QCOM, dtype, node_data,
-                                             shape)});
+        if (layout == CL_TENSOR_LAYOUT_OPTIMAL_QCOM) {
+          this->layer_.in_placeholder.insert(
+              {nid, MakeCLMLTensorFromJSONNode(node, CL_TENSOR_LAYOUT_NCHW_QCOM, dtype, node_data,
+                                               shape)});
+        } else {
+          this->layer_.in_placeholder.insert(
+              {nid, MakeCLMLTensorFromJSONNode(node, layout, dtype, node_data, shape)});
```

Comment:
Is it necessary to add an `ICHECK` to check the layout or not?

Reply:
This if else statement is to support other layout format CL_TENSOR_LAYOUT_NHWC_QCOM as well for input to the clml subgraph.